src/app/qgsversionmigration.cpp

```diff
@@ -53,14 +54,32 @@ QgsVersionMigration *QgsVersionMigration::canMigrate( int fromVersion, int toVer
 
 QgsError Qgs2To3Migration::runMigration()
 {
-  QgsError error;
+  QgsError errors;
   QgsError settingsErrors = migrateSettings();
   if ( !settingsErrors.isEmpty() )
   {
-    // TODO Merge error messages
+    for ( const auto &err : settingsErrors.messageList( ) )
+    {
+      errors.append( err );
+    }
   }
-  QgsError stylesError = migrateStyles();
-  return error;
+  QgsError stylesErrors = migrateStyles();
+  if ( !stylesErrors.isEmpty() )
+  {
+    for ( const auto &err : stylesErrors.messageList( ) )
+    {
+      errors.append( err );
+    }
+  }
+  QgsError authDbErrors = migrateAuthDb();
+  if ( !authDbErrors.isEmpty() )
+  {
+    for ( const auto &err : authDbErrors.messageList( ) )
```

Comment:
you'll need to make a const temporary copy of the list or stick to Q_FOREACH. Same above.

Reply:
@m-kuhn even if `messageList` is const (or at least that was it was mean to be) ?

Reply:
I thought so, but might really be that the const works here based on some research I did right now.

Even if this works, I would prefer to have this copy in place. The reason is, that if this is done wrong we have an expensive detach and detecting this in code reviews if at the place where it's used no copy is taken. The alternative would be that someone gets ground truth for the const return first and then modifies the whole code-source accordingly so we can assume this while reviewing.